tezos/new_context/src/working_tree/working_tree.rs

```diff
@@ -451,16 +451,38 @@ impl WorkingTree {
     }
 
     pub fn find_tree(&self, key: &ContextKey) -> Result<Option<Self>, MerkleError> {
+        let (file, path) = if let Some((file, path)) = key.split_last() {
+            (file, path)
+        } else {
+            // If the key is empty, we are checking self, which exists
+            return Ok(Some(self.clone()));
+        };
+
         let root = self.get_working_tree_root_ref();
         let mut storage = self.index.storage.borrow_mut();
 
-        let tree = self.find_raw_tree(root, key, &mut storage)?;
-
-        if tree.is_empty() {
-            return Ok(None);
+        if let Ok(tree_id) = self.find_raw_tree(root, path, &mut storage) {
+            if let Some(node_id) = storage.get_tree_node_id(tree_id, *file) {
+                let node = storage.get_node(node_id)?;
+                match node.get_entry() {
```

Comment:
@sebastiencs I need some help here because I may be misunderstanding how things are supposed to work here. I am getting a node as a result to the `get_node(node_id)` call, but then from `node.get_entry()` I get a `None`. What am I missing?

Reply:
Also, in this example if I comment `A` or `B` it works, but otherwise it doesn't. There seems to be an issue related to how the `storage` data is filled when a checkout is made.

```
    context = context.index.checkout(&context_hash_1)?.expect(&format!(
        "Commit not found: {}",
        context_hash_1.to_base58_check()
    ));
    // == A ==
    // context = context.add(
    //     &context_key!("data/rolls/owner/current/index/123"),
    //     &[1, 2, 3, 4, 5, 6, 7],
    // )?;
    // == B ==
    //assert!(context
    //    .find_tree(&context_key!("data/rolls/owner/current/index/123"))?
    //    .is_some());
    assert!(context
        .find_tree(&context_key!("data/rolls/owner/current/index"))?
        .is_some());
```

Reply:
If I do a `mem` on a children in that path that turns out to be a leaf (a value and not a directory), then `find_tree` also works as expected after that call.

Reply:
Ok, turns out I have to use `self.index.node_entry(node_id, &mut storage)`, it is that function that will try to retrieve the entry and put it in the storage.